templates/CRM/Profile/Form/Dynamic.tpl

```diff
@@ -40,7 +40,7 @@
       {$form._qf_Edit_upload_duplicate.html}
     </div>
   {/if}
-  {if $mode eq 1 || $activeComponent neq "CiviCRM"}
+  {if $mode eq 1 || empty($activeComponent) || $activeComponent neq "CiviCRM")}
```

Comment:
While smarty secure mode isn't working in v4+ yet, we've been avoiding empty and isset in tpls because of the way they get compiled when it is turned on, and so it ends up not fixing it. I would do this instead:

```diff
--- a/CRM/Core/BAO/UFGroup.php
+++ b/CRM/Core/BAO/UFGroup.php
@@ -900,6 +900,7 @@ class CRM_Core_BAO_UFGroup extends CRM_Core_DAO_UFGroup implements \Civi\Core\Ho
         $controller->run();

         $template = CRM_Core_Smarty::singleton();
+        $template->ensureVariablesAreAssigned(['activeComponent']);

         // Hide CRM error messages if they are set by the CMS.
         if (!empty($_POST)) {
```

Reply:
Thx I had a feeling I was missing something here!